src/ripple/net/impl/RPCCall.cpp

```diff
@@ -121,7 +121,8 @@ class RPCParser
     static Json::Value
     jvParseCurrencyIssuer(std::string const& strCurrencyIssuer)
     {
-        static boost::regex reCurIss("\\`([[:alpha:]]{3})(?:/(.+))?\\'");
+        static boost::regex reCurIss(
+            "\\`([][:alnum:]<>(){}[|?!@#$%^&*]{3})(?:/(.+))?\\'");
```

Comment:
How is this regex working correctly? Is the `tilde` character (third character from the beginning) replaceable with the single-quote (last but one character) inside the regex?

can we include a comment referring to [full alphabet](https://github.com/XRPLF/rippled/blob/5d011c7e6b399b988ad193d199e6e3dba92e49ad/src/ripple/protocol/impl/UintTypes.cpp#L33-L39) or reuse the `detail::isoCharSet` variable? [I was thinking: "\\`((detail::isoCharSet){3})(?:/(.+))?\\'", but I'm not sure]


also, I've used regex libraries in other languages, but this one baffles me. what is the purpose of the suffix of this regular expression? [`?:/(.+))?`]

Reply:
This is a `boost::regex`. It uses Perl syntax, which the library [documents](https://www.boost.org/doc/libs/1_82_0/libs/regex/doc/html/boost_regex/syntax/perl_syntax.html) online. Look at the section "[Buffer boundaries](https://www.boost.org/doc/libs/1_82_0/libs/regex/doc/html/boost_regex/syntax/perl_syntax.html#boost_regex.syntax.perl_syntax.buffer_boundaries)". (The character `` ` `` is called a [backtick](https://en.wikipedia.org/wiki/Backtick); [tilde](https://en.wikipedia.org/wiki/Tilde) is `~`.) `` \` `` matches the start of a buffer and `\'` matches the end of a buffer, so no, they cannot be replaced.

You could try to construct a regex string where `detail::isoCharSet` sits in a character class, but you would have to change the order within `detail::isoCharSet` because `]` cannot appear in the middle of a character class; it can only appear as the first character in the class. I don't think that factoring is worth it, though.

`(?:...)` is a non-capturing group. It groups subatoms into a larger atom without creating a capture group. So `(?:/(.+))?` optionally matches a forward slash and a non-empty sequence of characters, capturing the characters after the slash into a group.

Reply:
ok, this is helpful. thank you

Reply:
```suggestion
// The below regex matches the pattern as specified in UIntTypes.cpp:detail::isoCharSet. It is utilizes boost::regex and adheres to the Perl syntax for regular expressions.
            "\\`([][:alnum:]<>(){}[|?!@#$%^&*]{3})(?:/(.+))?\\'");
```

Reply:
@thejohnfreeman to accept, modify, or decline suggested comment

Reply:
I've pushed an alternative comment.